frontend/packages/gasless-transactions/src/hooks/use-fetch-voucher.ts

```diff
@@ -1,23 +1,24 @@
 import { useState, useEffect, useMemo, useCallback } from 'react';
 import { useVoucher, useBalanceFormat } from '@gear-js/react-hooks';
-import { ADDRESS } from '../consts';
-import { VOUCHER_MIN_LIMIT } from '@/features/wallet/consts';
+import { IS_CREATING_VOUCHER_ATOM, IS_UPDATING_VOUCHER_ATOM } from '../atoms';
+import { useAtom } from 'jotai';
+import { UseFetchVoucherProps } from '../types';
 
-export function useFetchVoucher(account: string | undefined) {
-  const { isVoucherExists, voucherBalance } = useVoucher(ADDRESS.GAME);
+export function useFetchVoucher({ accountAddress, programId, backendAddress, voucherLimit }: UseFetchVoucherProps) {
```

Comment:
Those arguments have the same origin across each app, I don't think that we should specify them explicitly. Better to do it in a convenient way in a package itself. As an example, you can look at `error-tracking` implementation.

- `accountAddress` is obtained from `useAccount` hook
- `programId` is usually env variable with default name `REACT_APP/VITE_CONTRACT_ADDRESS`
- `backendAddress` should be env variable with a default name too. My suggestion would be `GASLESS_BACKEND_ADDRESS`
- if `voucherLimit` preferrably would be controlled via env variable, same instructions as for above. I would suggest `GASLESS_BACKEND_VOUCHER_LIMIT`